server/aesdsocket.c

```diff
@@ -0,0 +1,341 @@
+		char *addr_ip = inet_ntoa(addr_in->sin_addr); // using inet_ntoa function
+
+		syslog(LOG_DEBUG, "Connection succesful. Accepting connection from %s", addr_ip);
+		printf("Connection succesful.Accepting connection from %s\n", addr_ip);
+
+		int i;
+		bool receive = false; // set the completion of packets to stop completion when
+
+		while (receive == false)
+		{
+
+			int temp_recv = recv(accept_fd, buff, BUFFER_SIZE, 0);
+			if (temp_recv < 0)
+			{
+				printf("Error while receving data packets\n");
+				syslog(LOG_ERR, "Error: Receiving failed =%s. Exiting ", strerror(errno));
+				exit(EXIT_FAILURE);
+			}
+			else if (temp_recv == 0)
+			{ // if no bytes received it will return 0
+				break;
+			}
+
+			for (i = 0; i < BUFFER_SIZE; i++)
+			{
+				if (buff[i] == '\n') // when \n received means end of file
+				{
+					receive = true; // make the bool true to end the receiving loop
+					i++;
+					printf("data packet receiving completed\n");
+					syslog(LOG_DEBUG, "data packet received");
+					break;
+				}
+			}
+			data_count += i; // increase data count for increasing packet size
+
+			// reallocate the buffer so as to accomodate the new data count
+			output_buffer = (char *)realloc(output_buffer, (data_count + 1));
+			if (output_buffer == NULL)
+			{
+				printf("buffer not created\n");
+				exit(EXIT_FAILURE);
+			}
+			// copy the buff to the output buffer of data count size
+			strncat(output_buffer, buff, i);
+			memset(buff, 0, BUFFER_SIZE); // memset buffer with 0
+		}
+		char *s = output_buffer; // to print the input characters
+		while (*s != '\n')
+		{
+			printf("%c", *s);
+			s++;
+		}
+		printf("\n");
+
+		// writing data to the file given in the path
+		file_fd = open(file_data, O_APPEND | O_WRONLY);
+		if (file_fd == -1) // generating errors
+		{
+			printf("Error: Opening file\n");
+			syslog(LOG_ERR, "Error: Opening file %s. Exiting ", strerror(errno));
+			exit(EXIT_FAILURE);
+		}
+
+		// Step-5 Writing to the file specified
+		int temp_write = write(file_fd, output_buffer, strlen(output_buffer));
+		if (temp_write == -1) // generating errors
+		{
+			printf("Error: File failed to be written \n");
+			syslog(LOG_ERR, "Error:  File failed to be written= %s. Exiting ", strerror(errno));
+			exit(EXIT_FAILURE);
+		}
+		else if (temp_write != strlen(output_buffer)) // generating errors if partially written
+		{
+			printf("Error: File not completely written\n");
+			syslog(LOG_ERR, "Error: File not completely written=%s. Exiting ", strerror(errno));
+			exit(EXIT_FAILURE);
+		}
+
+		close(file_fd);
+
+		// sending received data to the client back
+		memset(buff, 0, BUFFER_SIZE);
+		file_fd = open(file_data, O_RDONLY);
+		if (file_fd == -1) // generating errors
+		{
+			printf("Error: Opening file\n");
+			syslog(LOG_ERR, "Error: Opening file %s. Exiting ", strerror(errno));
+			exit(EXIT_FAILURE);
+		}
+
+		// Step-6 Reading from the file
+		send_buffer = (char *)malloc(sizeof(char) * (data_count + 1));
```

Comment:
@ayswariya088 what are you assuming here about the size of the process heap and the total size of all packets sent?